src/NHibernate/Impl/FilterImpl.cs

```diff
@@ -16,6 +17,7 @@ public class FilterImpl : IFilter
 		private FilterDefinition definition;
 
 		private readonly IDictionary<string, object> parameters = new Dictionary<string, object>();
+		private readonly IDictionary<string, IEnumerable> _parameterLists = new Dictionary<string, IEnumerable>();
```

Comment:
Why non generic `IEnumerable`?

Reply:
Because it is not possible to put anything more specific.

Reply:
`System.Array` is possible. But it will require to "materialize" the incoming collection.